File: GestorMaestro/GestorMaestro.Web/Modules/Escuela/Maestros/MaestrosColumns.cs
using MovieTutorial;
using Serenity.ComponentModel;
using System.ComponentModel;

namespace GestorMaestro.Escuela.Columns;

[ColumnsScript("Escuela.Maestros")]
[BasedOnRow(typeof(MaestrosRow), CheckNames = true)]
public class MaestrosColumns
{

    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int IdMaestro { get; set; }
    [EditLink]
    public string Nombre { get; set; }
    public string Apellido { get; set; }
  
    [DisplayName("Sexo")]

    public string IdSexoDescripcion { get; set; }
    public string Cedula { get; set; }
    
    [Width(500), MateriaListFormatter, QuickFilter ]
    [DisplayName("Materias")]

    public List<int> MateriaList { get; set; }

   

}

File: GestorMaestro/GestorMaestro.Web/Modules/ServerTypes/Escuela/MaestrosColumns.ts
import { MateriaListFormatter } from "@/Escuela/Maestros/MateriaListFormatter";
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { MaestrosRow } from "./MaestrosRow";

export interface MaestrosColumns {
    IdMaestro: Column<MaestrosRow>;
    Nombre: Column<MaestrosRow>;
    Apellido: Column<MaestrosRow>;
    IdSexoDescripcion: Column<MaestrosRow>;
    Cedula: Column<MaestrosRow>;
    MateriaList: Column<MaestrosRow>;
}

export class MaestrosColumns extends ColumnsBase<MaestrosRow> {
    static readonly columnsKey = 'Escuela.Maestros';
    static readonly Fields = fieldsProxy<MaestrosColumns>();
}

[MateriaListFormatter]; // referenced types